mantidproject/mantid
# DetectorInfo: sync scan support for better performance

Add methods for setting up `DetectorInfo` for a sync scan of all detectors. This avoid a big overhead when merging many `DetectorInfo` objects into one big scan.

See also https://github.com/mantidproject/mantid/pull/20204 (replaced by this PR).

**To test:**

Code review.

Try loading the attached file for D20 with and without the changes, preferably on a release branch. Check the speed up (on my PC 15.8s without changes, 3s with).

[d20_det_scan.zip](https://github.com/mantidproject/mantid/files/1211628/d20_det_scan.zip)

Fixes #20027.

**To test:**

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
